modules/consul/src/test/java/org/testcontainers/consul/ConsulContainerTest.java

```diff
@@ -0,0 +1,78 @@
+package org.testcontainers.consul;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.Response;
+import com.ecwid.consul.v1.kv.model.GetValue;
+import io.restassured.RestAssured;
+import org.assertj.core.api.Assertions;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This test shows the pattern to use the ConsulContainer @ClassRule for a junit test. It also has tests that ensure
+ * the properties were added correctly by reading from Consul with the CLI and over HTTP.
+ */
+public class ConsulContainerTest {
+
+    @ClassRule
+    public static ConsulContainer consulContainer = new ConsulContainer(ConsulTestImages.CONSUL_IMAGE)
+        .withConsulCommand("kv put config/testing1 value123");
+
+    @Test
+    public void readFirstPropertyPathWithCli() throws IOException, InterruptedException {
+        GenericContainer.ExecResult result = consulContainer.execInContainer("consul", "kv", "get", "config/testing1");
```

Comment:
I think `withConsulCommand` can be used here too

```suggestion
        GenericContainer.ExecResult result = consulContainer.withConsulCommand("kv get config/testing1");
```